tests/ignite/contrib/metrics/test_precision_recall_curve.py

```diff
@@ -124,3 +135,182 @@ def test_check_compute_fn():
 
     em = PrecisionRecallCurve(check_compute_fn=False)
     em.update(output)
+
+
+def _test_distrib_compute(device):
+
+    rank = idist.get_rank()
+    torch.manual_seed(12)
+
+    def _test(y_pred, y, batch_size, metric_device):
+
+        metric_device = torch.device(metric_device)
+        prc = PrecisionRecallCurve(device=metric_device)
+
+        torch.manual_seed(10 + rank)
+
+        prc.reset()
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                prc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+        else:
+            prc.update((y_pred, y))
+
+        # gather y_pred, y
+        y_pred = idist.all_gather(y_pred)
+        y = idist.all_gather(y)
+
+        np_y = y.cpu().numpy()
+        np_y_pred = y_pred.cpu().numpy()
+
+        res = prc.compute()
+
+        assert isinstance(res, Tuple)
+        assert precision_recall_curve(np_y, np_y_pred)[0] == pytest.approx(res[0])
+        assert precision_recall_curve(np_y, np_y_pred)[1] == pytest.approx(res[1])
+        assert precision_recall_curve(np_y, np_y_pred)[2] == pytest.approx(res[2])
+
+    def get_test_cases():
+        test_cases = [
+            # Binary input data of shape (N,) or (N, 1)
+            (torch.randint(0, 2, size=(10,)), torch.randint(0, 2, size=(10,)), 1),
+            (torch.randint(0, 2, size=(10, 1)), torch.randint(0, 2, size=(10, 1)), 1),
+            # updated batches
+            (torch.randint(0, 2, size=(50,)), torch.randint(0, 2, size=(50,)), 16),
+            (torch.randint(0, 2, size=(50, 1)), torch.randint(0, 2, size=(50, 1)), 16),
+        ]
+        return test_cases
+
+    for _ in range(5):
+        test_cases = get_test_cases()
+        for y_pred, y, batch_size in test_cases:
+            _test(y_pred, y, batch_size, "cpu")
+            if device.type != "xla":
+                _test(y_pred, y, batch_size, idist.device())
+
+
+def _test_distrib_integration(device):
+
+    rank = idist.get_rank()
```

Comment:
This test case does not use rank and each process generates some random preds and true values. Implementation of `PrecisionRecallCurve` should gather all data from all processes but it is checked against local process computation:
```
        np_y_true = y_true.cpu().numpy().ravel()
        np_y_preds = y_preds.cpu().numpy().ravel()
        sk_precision, sk_recall, sk_thresholds = precision_recall_curve(np_y_true, np_y_preds)
```

I would assume this to fail but looks like it is passing. @sayantan1410 can you check why it is so ?

Reply:
Yeah sur will check it !!

Reply:
Actually the error comes from the following which seems wrong 
https://github.com/pytorch/ignite/blob/90d79aa7b6d0a52b2741943b62baab2ea0461453/tests/ignite/contrib/metrics/regression/test_median_absolute_error.py#L152

@sayantan1410 Have a look to this correct implementation
https://github.com/pytorch/ignite/blob/90d79aa7b6d0a52b2741943b62baab2ea0461453/tests/ignite/contrib/metrics/regression/test_mean_error.py#L139

Reply:
@sdesrozis A small question, 
https://github.com/pytorch/ignite/blob/90d79aa7b6d0a52b2741943b62baab2ea0461453/tests/ignite/contrib/metrics/regression/test_mean_error.py#L139

Here, also we don't have something like idist.all_gather so how is the data getting gathered from all the processes ?